pkg/services/team/teamimpl/team.go

```diff
@@ -3,65 +3,129 @@ package teamimpl
 import (
 	"context"
 
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+
 	"github.com/grafana/grafana/pkg/infra/db"
+	"github.com/grafana/grafana/pkg/infra/tracing"
 	"github.com/grafana/grafana/pkg/services/team"
 	"github.com/grafana/grafana/pkg/setting"
 )
 
 type Service struct {
-	store store
+	store  store
+	tracer tracing.Tracer
 }
 
-func ProvideService(db db.DB, cfg *setting.Cfg) (team.Service, error) {
+func ProvideService(db db.DB, cfg *setting.Cfg, tracer tracing.Tracer) (team.Service, error) {
 	store := &xormStore{db: db, cfg: cfg, deletes: []string{}}
 
 	if err := store.uidMigration(); err != nil {
 		return nil, err
 	}
-	return &Service{store: &xormStore{db: db, cfg: cfg, deletes: []string{}}}, nil
+	return &Service{
+		store:  &xormStore{db: db, cfg: cfg, deletes: []string{}},
+		tracer: tracer,
+	}, nil
 }
 
 func (s *Service) CreateTeam(name, email string, orgID int64) (team.Team, error) {
+	_, span := s.tracer.Start(context.Background(), "team.CreateTeam", trace.WithAttributes(
```

Comment:
I think this and the `IsTeamMember` functions would create a new root span (because the context is not propagated).I'm not 100% sure, but the current context should be passed to those functions to have proper tracing. Wdyt?

It can be also handled in a follow up PR.

Reply:
Thanks for pointing it out. I was putting it off as `CreateTeam` is used in soooo many tests but otherwise only from the team creation API function, so it felt like a lot of work for no benefit (as no useful context would be propagated for any non-test code execution path). But now that I see the amount of test changes that are required for `ProvideService` anyway, I might as well add a few more 😄